scrapy_poet/page_input_providers.py

```diff
@@ -181,4 +188,13 @@ def serialize(self, result: Sequence[Any]) -> Any:
         return [attr.asdict(response_data) for response_data in result]
 
     def deserialize(self, data: Any) -> Sequence[Any]:
-        return [ResponseData(**response_data) for response_data in data]
+        return [
+            HttpResponse(
+                response_data["url"],
+                response_data["body"],
+                status=response_data["status"],
+                headers=HttpResponseHeaders.from_bytes_dict(response_data["headers"]),
```

Comment:
Why is HttpResponseHeaders.from_bytes_dict needed here?

Reply:
@kmike It's because Scrapy headers look like this:

```python
scrapy_headers = {
    b"Content-Encoding": [b"gzip", b"br"],
    b"Content-Type": [b"text/html"],
    b"content-length": b"648",
}
```

Reference for the motivation: https://github.com/scrapinghub/web-poet/pull/33#issue-1183279271

Reply:
If I'm not wrong, here the result is not coming from Scrapy, it's a part of serialization / deserialization for cache.

Reply:
Ahhh I see what you mean. Nice catch @kmike ! 🙌 Fixed this in https://github.com/scrapinghub/scrapy-poet/pull/72